WordPress/gutenberg
# Add a pre-commit hook to check whether API docs are updated

This PR is an alternative to https://github.com/WordPress/gutenberg/pull/18657 Instead of automatically adding files to the commit (which [could be problematic](https://github.com/WordPress/gutenberg/pull/18735#issuecomment-559018142)) this goes the route of checking whether the API docs are unstaged: if so, the commit will be prevented and the user has the choice of either adding them to the commit or skipping the check.

## Test instructions

* Edit the JSON comment of an exported entity in `packages/core-data/src/actions.js`
* `git add packages/core-data/src/actions.js`
* `git commit -m 'testing'`

The expected result is that you can't commit due to the unstaged README files that were auto-updated due to the code change and will see a message like:

![Screenshot from 2019-12-03 10-33-09](https://user-images.githubusercontent.com/583546/70039137-b51e0d00-15b9-11ea-872b-2f2d1a98113b.png)
